### ISSUE TITLE
[Bug]: Persistent database queries: `UPDATE oc_authtoken SET password`

### ISSUE THREAD

**Issue by taiBsu** [nextcloud/server#33216]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

I am having persistent database queries when executing `show processlist;` in my MariaDB:
```
| 25886 | tn_nextcloud   | nextcloud.*:47428   | nextcloud      | Query   |     0 | Updating status | UPDATE `oc_authtoken` SET `password` = '******** |    0.000 |
| 25887 | tn_nextcloud   | nextcloud.*:47430   | nextcloud      | Query   |     1 | Updating status | UPDATE `oc_authtoken` SET `password` = '******** |    0.000 |
| 25933 | tn_nextcloud   | nextcloud.*:47460   | nextcloud      | Query   |     0 | Updating status | UPDATE `oc_authtoken` SET `password` = '******** |    0.000 |
| 25934 | tn_nextcloud   | nextcloud.*:47462   | nextcloud      | Query   |     0 | Updating status | UPDATE `oc_authtoken` SET `password` = '******** |    0.000 |
| 25935 | tn_nextcloud   | nextcloud.*:47466   | nextcloud      | Query   |     0 | Updating status | UPDATE `oc_authtoken` SET `password` = '******** |    0.000 |
| 25937 | tn_nextcloud   | nextcloud.*:47470   | nextcloud      | Query   |     0 | Updating status | UPDATE `oc_authtoken` SET `password` = '******** |    0.000 |
| 25938 | tn_nextcloud   | nextcloud.*:47474   | nextcloud      | Query   |     2 | Commit          | UPDATE `oc_authtoken` SET `password` = '******** |    0.000 |
| 25939 | tn_nextcloud   | nextcloud.*:47476   | nextcloud      | Query   |     0 | Updating status | UPDATE `oc_authtoken` SET `password` = '******** |    0.000 |
| 26033 | tn_nextcloud   | nextcloud.*:47546   | nextcloud      | Query   |     0 | Updating status | UPDATE `oc_authtoken` SET `password` = '******** |    0.000 |
| 26034 | tn_nextcloud   | nextcloud.*:47548   | nextcloud      | Query   |     0 | Updating status | UPDATE `oc_authtoken` SET `password` = '******** |    0.000 |
| 26035 | tn_nextcloud   | nextcloud.*:47550   | nextcloud      | Query   |     0 | Commit          | UPDATE `oc_authtoken` SET `password` = '******** |    0.000 |
| 26036 | tn_nextcloud   | nextcloud.*:47554   | nextcloud      | Query   |     0 | Updating status | UPDATE `oc_authtoken` SET `password` = '******** |    0.000 |
| 26037 | tn_nextcloud   | nextcloud.*:47556   | nextcloud      | Query   |     0 | Updating status | UPDATE `oc_authtoken` SET `password` = '******** |    0.000 |
| 26038 | tn_nextcloud   | nextcloud.*:47558   | nextcloud      | Query   |     0 | Updating status | UPDATE `oc_authtoken` SET `password` = '******** |    0.000 |
| 26039 | tn_nextcloud   | nextcloud.*:47560   | nextcloud      | Query   |     0 | Updating status | UPDATE `oc_authtoken` SET `password` = '******** |    0.000 |
| 26040 | tn_nextcloud   | nextcloud.*:47562   | nextcloud      | Query   |     0 | Updating status | UPDATE `oc_authtoken` SET `password` = '******** |    0.000 |
| 26043 | tn_nextcloud   | nextcloud.*:47564   | nextcloud      | Query   |     0 | Updating status | UPDATE `oc_authtoken` SET `password` = '******** |    0.000 |
```

Is this normal behavior? Those queries are being displayed almost permanently.

Thanks in advance.

### Steps to reproduce

1. Start nextcloud container
2. Open MariaDB console
3. `show processlist;`

### Expected behavior

Not quite sure what the expected behavior would be here.

### Installation method

Official Docker image

### Operating system

Debian/Ubuntu

### Web server

Apache (supported)

### Database engine version

MariaDB

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "taibsu.net",
            "www.taibsu.net",
            "luke",
            "localhost",
            "web",
            "*",
            "cloud.taibsu.net"
        ],
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "overwritehost": "cloud.taibsu.net",
        "overwriteprotocol": "https",
        "overwritewebroot": "\/",
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "24.0.2.1",
        "overwrite.cli.url": "https:\/\/cloud.taibsu.net\/",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "mail_smtpmode": "smtp",
        "mail_smtpsecure": "ssl",
        "mail_sendmailmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "maintenance": false,
        "theme": "",
        "logfile": "\/var\/log\/nextcloud.log",
        "loglevel": 2,
        "log_rotate_size": "104857600",
        "updater.release.channel": "stable",
        "mysql.utf8mb4": true,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "app_install_overwrite": [
            "spreed",
            "social"
        ],
        "data-fingerprint": "b97a38a37134555daa2d5fd54e058302",
        "preview_max_x": "2048",
        "preview_max_y": "2048",
        "jpeg_quality": "60",
        "default_phone_region": "DE"
    }
}
```

### List of activated Apps

```shell
Enabled:
  - accessibility: 1.10.0
  - admin_audit: 1.14.0
  - bookmarks: 10.5.1
  - calendar: 3.4.2
  - cloud_federation_api: 1.7.0
  - comments: 1.14.0
  - contacts: 4.1.1
  - contactsinteraction: 1.5.0
  - cookbook: 0.9.13
  - cospend: 1.4.8
  - dashboard: 7.4.0
  - dav: 1.22.0
  - deck: 1.7.1
  - federatedfilesharing: 1.14.0
  - federation: 1.14.0
  - files: 1.19.0
  - files_markdown: 2.3.6
  - files_pdfviewer: 2.5.0
  - files_rightclick: 1.3.0
  - files_sharing: 1.16.2
  - files_trashbin: 1.14.0
  - files_videoplayer: 1.13.0
  - firstrunwizard: 2.13.0
  - integration_github: 1.0.2
  - logreader: 2.9.0
  - lookup_server_connector: 1.12.0
  - mail: 1.13.6
  - news: 18.1.0
  - notes: 4.4.0
  - notifications: 2.12.0
  - oauth2: 1.12.0
  - password_policy: 1.14.0
  - passwords: 2022.6.20
  - photos: 1.6.0
  - polls: 3.7.0
  - previewgenerator: 5.0.0
  - privacy: 1.8.0
  - provisioning_api: 1.14.0
  - quicknotes: 0.8.0
  - recommendations: 1.3.0
  - serverinfo: 1.14.0
  - settings: 1.6.0
  - sharebymail: 1.14.0
  - support: 1.7.0
  - survey_client: 1.12.0
  - systemtags: 1.14.0
  - tasks: 0.14.4
  - text: 3.5.1
  - theming: 1.15.0
  - twofactor_backupcodes: 1.13.0
  - updatenotification: 1.14.0
  - user_status: 1.4.0
  - viewer: 1.8.0
  - weather_status: 1.4.0
  - workflowengine: 2.6.0
Disabled:
  - activity: 2.15.0
  - circles: 0.16.0
  - encryption: 2.4.0
  - files_external
  - files_versions: 1.11.0
  - nextcloud_announcements: 1.13.0
  - user_ldap: 1.13.1
```

### Nextcloud Signing status

```shell
No errors have been found.
```

**Comment by andyxheli** [comment]
I did the same thing and my looks normal 
![image](https://user-images.githubusercontent.com/59488153/179037550-fda7d1e0-bf15-4ce3-bfef-065d4d890cbd.png)

**Comment by tcitworld** [comment]
I have the same issue with PostgreSQL, on NC 24.0.3 (and previously 24.0.2). It's becoming a real issue as this generates a lot of WAL changes which are used for backups (3 times the size of the whole database - hundreds of gigabytes - each day).

```
my_db=# SELECT pid, age(clock_timestamp(), query_start), usename, query 
FROM pg_stat_activity 
WHERE query != '<IDLE>' AND query NOT ILIKE '%pg_stat_activity%' AND query like '%oc_authtoken%' 
ORDER BY query_start desc;
   pid   |       age       | usename  |                           query                           
---------+-----------------+----------+-----------------------------------------------------------
 1112197 | 00:00:00.000641 | my_user | UPDATE "oc_authtoken" SET "password" = $1 WHERE "id" = $2
 1111455 | 00:00:00.000648 | my_user | UPDATE "oc_authtoken" SET "password" = $1 WHERE "id" = $2
 1112773 | 00:00:00.000688 | my_user | UPDATE "oc_authtoken" SET "password" = $1 WHERE "id" = $2
 1112123 | 00:00:00.001281 | my_user | UPDATE "oc_authtoken" SET "password" = $1 WHERE "id" = $2
 1113816 | 00:00:00.001328 | my_user | UPDATE "oc_authtoken" SET "password" = $1 WHERE "id" = $2
 1113914 | 00:00:00.001378 | my_user | UPDATE "oc_authtoken" SET "password" = $1 WHERE "id" = $2
```

This is the trace I was able to find from `OC\Authentication\Token\PublicKeyToken::setPassword`, which seems to cause all these database updates:
```
{"file":"/var/www/nextcloud/lib/public/AppFramework/Db/Entity.php","line":73,"function":"setPassword","class":"OC\\Authentication\\Token\\PublicKeyToken","type":"->"},
{"file":"/var/www/nextcloud/lib/private/Authentication/Token/PublicKeyTokenMapper.php","line":91,"function":"fromRow","class":"OCP\\AppFramework\\Db\\Entity","type":"::"},
{"file":"/var/www/nextcloud/lib/private/Authentication/Token/PublicKeyTokenProvider.php","line":111,"function":"getToken","class":"OC\\Authentication\\Token\\PublicKeyTokenMapper","type":"->","args":["*** sensitive parameters replaced ***"]},
{"file":"/var/www/nextcloud/lib/private/Authentication/Token/Manager.php","line":133,"function":"getToken","class":"OC\\Authentication\\Token\\PublicKeyTokenProvider","type":"->","args":["*** sensitive parameters replaced ***"]},
{"file":"/var/www/nextcloud/lib/private/User/Session.php","line":774,"function":"getToken","class":"OC\\Authentication\\Token\\Manager","type":"->","args":["*** sensitive parameters replaced ***"]},
{"file":"/var/www/nextcloud/lib/private/User/Session.php","line":257,"function":"validateToken","class":"OC\\User\\Session","type":"->","args":["*** sensitive parameters replaced ***"]},
{"file":"/var/www/nextcloud/lib/private/User/Session.php","line":232,"function":"validateSession","class":"OC\\User\\Session","type":"->"},
{"file":"/var/www/nextcloud/lib/private/legacy/OC_App.php","line":390,"function":"getUser","class":"OC\\User\\Session","type":"->"},
{"file":"/var/www/nextcloud/lib/private/legacy/OC_App.php","line":122,"function":"getEnabledApps","class":"OC_App","type":"::"},
{"file":"/var/www/nextcloud/lib/private/legacy/OC_User.php","line":127,"function":"loadApps","class":"OC_App","type":"::"},
{"file":"/var/www/nextcloud/lib/base.php","line":727,"function":"setupBackends","class":"OC_User","type":"::"},
{"file":"/var/www/nextcloud/lib/base.php","line":1104,"function":"init","class":"OC","type":"::"},
{"file":"/var/www/nextcloud/ocs/v1.php","line":31,"args":["/var/www/nextcloud/lib/base.php"],"function":"require_once"},
{"file":"/var/www/nextcloud/ocs/v2.php","line":23,"args":["/var/www/nextcloud/ocs/v1.php"],"function":"require_once"}
```

I'm wondering if when the `fromRow` from `getToken` calls the `setPassword` setter it doesn't at the same time trigger an update trough the DB entity.

**Comment by tcitworld** [comment]
Another related issue https://github.com/nextcloud/server/issues/33063

**Comment by not-implemented** [comment]
see #33485

**Comment by taiBsu** [comment]
> see #33485

Nice, thanks. So it looks like this would be fixed in 25.0?

**Comment by not-implemented** [comment]
seems it is already in 24.0.4 stable ... I updated today and the change is already in the installed source :-)
